pkg/volume/csi/csi_mounter.go

```diff
@@ -107,49 +111,97 @@ func (c *csiMountMgr) SetUpAt(dir string, fsGroup *int64) error {
 		return nil
 	}
 
-	csiSource, err := getCSISourceFromSpec(c.spec)
-	if err != nil {
-		klog.Error(log("mounter.SetupAt failed to get CSI persistent source: %v", err))
-		return err
-	}
-
 	csi := c.csiClient
 	ctx, cancel := context.WithTimeout(context.Background(), csiTimeout)
 	defer cancel()
 
-	// Check for STAGE_UNSTAGE_VOLUME set and populate deviceMountPath if so
-	deviceMountPath := ""
-	stageUnstageSet, err := csi.NodeSupportsStageUnstage(ctx)
+	volSrc, pvSrc, err := getSourceFromSpec(c.spec)
```

Comment:
nit: `getCSISourceFromSpec`

Reply:
Yes, I will overhaul the func names for retrieving volume sources.